doc/wiki/3_iiQKA.OS2_RSI.md

```diff
@@ -98,20 +98,101 @@ The parameters in the driver configuration file can be also changed during runti
+
+1. The `rsi_gpio_joint_pos.rsix` file offers an example of how to set up the different I/Os. For detailed instructions, please refer to the RSI manual on KUKA Xpert.
+   - The file can be edited via RSI Visual in iiQWorks.Sim.
+   - All I/Os should be connected to the inputs or outputs of the Ethernet RSI object.
+2. To run the GPIO example, a `rsi_gpio_example.src` file has been added.
+
+Since both iiQKA.OS2 and KSS systems uses the same message format, the Ethernet object's `xml` file can be found in the `kuka_external_control_sdk/kss/krl/SensorInterface` directory:
+
+1. The Ethernet RSI object requires the `rsi_gpio_ethernet.xml` configuration file.
+   - The `<SEND>` object contains all parameters that are sent to the client.
+   - The `<RECEIVE>` object contains all the parameters that are received from the client.
+   - To add a new I/O element the following parameters must be set:
+     - `TAG`: Contains the aforementioned unique key. The tag format is: `GPIO.UniqueKey`. It must start with `GPIO`, followed by a `.`, and then the `key`.
+     - `TYPE`: The type can be one of the following: `BOOL`, `DOUBLE`, or `LONG`.
+     - `INDX`: This must match the configuration of the I/O object in RSI Visual for the Ethernet object. This is the only parameter that connects the XML file entries to those in the `.rsix` file.
+     - `HOLDON`: This is only used in the `<RECEIVE>` object. It sets the behavior of the output when packets are missed:
+       - `0`: The output is reset.
+       - `1`: The most recent valid value remains at the output.
+   - A sample configuration for one I/O element:
+
+      ```xml
+      <ELEMENT TAG="GPIO.OUTPUT_01" TYPE="DOUBLE" INDX="1" HOLDON="1" />
+      ```
+
+###### Client side configuration
+
+To configure the client side, two configuration files need to be completed:
+
+1. The `kuka_rsi_driver/config/gpio_config.xacro` file is an extension to the robot's URDF and contains a `<gpio>` tag as part of the ROS Control parameters.
+   - The GPIO object must be called `gpio`.
+   - The state and command interfaces must be configured according to the example provided in the file.
+   - For each interface, several additional parameters are available:
+     - `name`: The previously mentioned unique key.
+     - `data_type`: The data type of the interface. Must be one of the three supported by RSI: `BOOL`, `DOUBLE`, or `LONG`.
+     - `limits`: Enables additional limit checking (defaults to `true`).
+     - `min`: Minimum value for limit checking. (If not used or used incorrectly, `limits` is set to `false`.)
+     - `max`: Maximum value for limit checking. (If not used or used incorrectly, `limits` is set to `false`.)
+     - `initial_value`: Initial value of the interface. Mostly useful for outputs without a state interface, as this value is otherwise overridden during the first cycle.
```

Comment:
```suggestion
     - `initial_value`: Initial value of the interface. The value must be in a number format for every data type. Mostly useful for outputs without a state interface, as this value is otherwise overridden during the first cycle.